src/Symfony/Component/Serializer/Normalizer/ObjectNormalizer.php

```diff
@@ -0,0 +1,162 @@
+use Symfony\Component\Serializer\Exception\LogicException;
+use Symfony\Component\Serializer\Mapping\Factory\ClassMetadataFactory;
+use Symfony\Component\Serializer\NameConverter\NameConverterInterface;
+
+/**
+ * Converts between objects and arrays using the PropertyAccess component.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+class ObjectNormalizer extends AbstractNormalizer
+{
+    /**
+     * @var PropertyAccessorInterface
+     */
+    protected $propertyAccessor;
+
+    public function __construct(ClassMetadataFactory $classMetadataFactory = null, NameConverterInterface $nameConverter = null, PropertyAccessorInterface $propertyAccessor = null)
+    {
+        parent::__construct($classMetadataFactory, $nameConverter);
+
+        $this->propertyAccessor = $propertyAccessor ?: PropertyAccess::createPropertyAccessor();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supportsNormalization($data, $format = null)
+    {
+        return is_object($data);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws CircularReferenceException
+     */
+    public function normalize($object, $format = null, array $context = array())
+    {
+        if ($this->isCircularReference($object, $context)) {
+            return $this->handleCircularReference($object);
+        }
+
+        $data = array();
+        $attributes = $this->getAllowedAttributes($object, $context);
+
+        // If not using groups, detect manually
+        if (false === $attributes) {
+            $attributes = array();
+
+            // methods
+            $reflClass = new \ReflectionClass($object);
+            foreach ($reflClass->getMethods(\ReflectionMethod::IS_PUBLIC) as $reflMethod) {
+                if (
+                    !$reflMethod->isConstructor() &&
+                    !$reflMethod->isDestructor() &&
+                    0 === $reflMethod->getNumberOfRequiredParameters()
+                ) {
+                    $name = $reflMethod->getName();
+
+                    if (strpos($name, 'get') === 0 || strpos($name, 'has') === 0) {
+                        // getters and hassers
+                        $attributes[lcfirst(substr($name, 3))] = true;
+                    } elseif (strpos($name, 'is') === 0) {
+                        // issers
+                        $attributes[lcfirst(substr($name, 2))] = true;
+                    }
+                }
+            }
+
+            // properties
+            foreach ($reflClass->getProperties(\ReflectionProperty::IS_PUBLIC) as $reflProperty) {
+                $attributes[$reflProperty->getName()] = true;
+            }
+
+            $attributes = array_keys($attributes);
+        }
+
+        foreach ($attributes as $attribute) {
+            if (in_array($attribute, $this->ignoredAttributes)) {
+                continue;
+            }
+
+            $attributeValue = $this->propertyAccessor->getValue($object, $attribute);
```

Comment:
Ok I got it. But still no because the attribute value will never be retrieved. We need it even if a callback is not registered.